mail_tracking/controllers/main.py

```diff
@@ -68,14 +68,13 @@ def mail_tracking_open(self, db, tracking_email_id, token=False, **kw):
             try:
                 tracking_email = env['mail.tracking.email'].search([
                     ('id', '=', tracking_email_id),
-                    ('state', 'in', ['sent', 'delivered']),
                     ('token', '=', token),
                 ])
-                if tracking_email:
-                    tracking_email.event_create('open', metadata)
-                else:
+                if not tracking_email:
                     _logger.warning(
                         "MailTracking email '%s' not found", tracking_email_id)
+                elif tracking_email.state in ('sent', 'delivered'):
```

Comment:
```suggestion
                elif tracking_email.state in {'sent', 'delivered'}:
```

Reply:
Sure, but may I ask why is this better?

Reply:
In this exactly case... tuples are better, sorry.

But in general, sets are better for search purposes:
```timeit.timeit('"two" in data', number=10000, setup='data=("one", "two")')```
```timeit.timeit('"two" in data', number=10000, setup='data={"one", "two"}')```

Reply:
Ah ok, tnx. Did not know this.